src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -503,8 +520,60 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
+                    FunctionResult functionResult = null;
+
+                    // Create a dictionary that will pass the user properties from the Executing filter to the Executed filter
+                    Dictionary<string, object> properties = new Dictionary<string, object>();
+
+                    // Create the context objects for the filter execution
+                    FunctionExecutingContext executingContext = new FunctionExecutingContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger);
+                    executingContext.JobHost = jobHost;
+                    executingContext.Config = config;
+
+                    FunctionExecutedContext executedContext = new FunctionExecutedContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger, functionResult);
+                    executedContext.JobHost = jobHost;
+                    executedContext.Config = config;
+
+                    // Lambda function for function invocation
+                    Func<Task> invokeAsync = async () =>
+                    {
+                        await InvokeAsync(invoker, parameterHelper, timeoutTokenSource, functionCancellationTokenSource,
+                            throwOnTimeout, timerInterval, instance);
+                    };
+
+                    // Lambda function that wraps filters around the function
+                    Func<Func<Task>, IFunctionInvocationFilter, Func<Task>> wrapFunctionWithFilter = (innerFunction, filter) =>
+                    {
+                        Func<Task> filteredFunction = async () =>
+                        {
+                            try
+                            {
+                                await filter.OnExecutingAsync(executingContext, functionCancellationTokenSource.Token);
+                                await innerFunction();
+                                await filter.OnExecutedAsync(executedContext, functionCancellationTokenSource.Token);
+                            }
+                            catch (Exception e)
+                            {
+                                // How should the exception results be cached?
+                                executedContext.Result = new FunctionResult(e);
+                            }
+                        };
+                        return filteredFunction;
+                    };
+                    
+                    Func<Task> filteredInvocation = filters?.Aggregate(invokeAsync, wrapFunctionWithFilter);
+
+                    await filteredInvocation();
+
+                    // Throw for the sake of error tracing
+                    if (executedContext.Result != null)
+                    {
+                        if (!executedContext.Result.Succeeded)
+                        {
+                            throw executedContext.Result.Exception;
```

Comment:
Instead of re-throwing here which destroys the call stack, use an [ExceptionDispatchInfo](https://msdn.microsoft.com/en-us/library/system.runtime.exceptionservices.exceptiondispatchinfo.capture(v=vs.110).aspx) to capture the exception inside the filteredFunction lambda and then throw (which maintains call stack)

This may change depending on the AggregateException behavior above